libnode.yml

```diff
@@ -0,0 +1,71 @@
+# Build Node.js as a dynamic library for use in the Node.js binding
+
+trigger: none
+pr: none
+jobs:
+
+################################################################################
+- job: ubuntu_1604
+################################################################################
+  displayName: Ubuntu 16.04
+  pool:
+    vmImage: ubuntu-16.04
+  steps:
+  - script: |
+      v1/tools/build_nodejs.sh
+      # Convert thin archives to normal ones
+      libs=`find v1/build_nodejs/dist/lib/ -name '*.a'`
+      for lib in $libs; do
+        ar -t $lib | xargs ar -rs $lib.tmp && mv $lib.tmp $lib
```

Comment:
Curious, what's this command doing?

Reply:
Ah, right. When you use `./configure --shared` to set up the Node.js build, it builds the static libraries as thin archives: they still reference the individual object files (which were not copied into `dist/lib`). That's fine when you turn around and consume those libraries in another build on the same machine, but I needed to publish them as artifacts that other builds (other machines) can consume. I used these commands to make the libraries self-contained. The first `ar` command lists all the object files referenced by a given {lib}.a, and the second command inserts them into the library and updates the index.

Reply:
Interesting. So just having the `.so` file is not sufficient? Didn't know that!